### PR TITLE
Test Framework support for interactive login

### PR DESCRIPTION
**Pull request by shahabhijeet**
- Adding Interactive login for environment that still supports it.
- Adding support for Custom environment, this will enable targeting any environment as long as you provide all endpoints in the connection string.
- Adding tests. These are live tests hence are skipped, because it needs interactive login.
- Updates for VS2017

### CODE REVIEW

File: src/TestFramework/Microsoft.Rest.ClientRuntime.Azure.TestFramework/TestEndpoints.cs [hunk 1]
```diff
@@ -212,5 +260,9 @@ internal TestEndpoints(TestEndpoints testEndpoint, ConnectionString connString):
         // Graph Token Audience 
         public Uri GraphTokenAudienceUri { get; set; }
 
+        public Uri PublishingSettingsFileUri { get; set; }
+
+        //public Uri AADServiceUri { get; set; }
```
Comment (markcowl) [inline src/TestFramework/Microsoft.Rest.ClientRuntime.Azure.TestFramework/TestEndpoints.cs]: Please remove, do not comment this out.

File: src/TestFramework/Microsoft.Rest.ClientRuntime.Azure.TestFramework/TestEnvironment.cs [hunk 2]
```diff
@@ -322,29 +348,70 @@ private void InteractiveLogin(string tenant, ActiveDirectoryServiceSettings aadS
...
+            }
+            else
+            {
+                // The idea is in case if no match was found, we check if subscription Id was provided in connection string, if not
+                // we then check if the retrieved subscription list has exactly 1 subscription, if yes we will just use that one.
+                if (string.IsNullOrEmpty(this.SubscriptionId))
+                {
+                    if (subscriptionList.Count() == 1)
+                    {
+                       this.ConnectionString.KeyValuePairs[ConnectionStringKeys.SubscriptionIdKey] = subscriptionList.First<SubscriptionInfo>().SubscriptionId;
```
Comment (markcowl) [inline src/TestFramework/Microsoft.Rest.ClientRuntime.Azure.TestFramework/TestEnvironment.cs]: Please file an issue to add a test for this scenario.  We should be able to mock the APIs to make this possible to test.

**Review by markcowl — CHANGES_REQUESTED** [review]

File: src/TestFramework/TestFramework.Net45Tests/TestEnvironment/TestEnvTests.cs [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                "ServiceManagementUri=https://management.core.cloudapi.de/;" +
+                "GalleryUri=https://gallery.cloudapi.de/;" +
+                "GraphUri=https://graph.cloudapi.de/;" +
+                "AADAuthUri=https://login.microsoftonline.de/;" +
+                "IbizaPortalUri=http://portal.microsoftazure.de/;" +
+                "RdfePortalUri=https://management.core.cloudapi.de/;" +
+                "GraphTokenAudienceUri=https://graph.cloudapi.de/;" +
+                "AADTokenAudienceUri=https://management.core.cloudapi.de/"
+            )]
+        public void InteractiveGermanLoginNoSubscription(string connStr)
```
Comment (shahabhijeet) [inline src/TestFramework/TestFramework.Net45Tests/TestEnvironment/TestEnvTests.cs]: @markcowl Test is already added for that scenario.

File: src/TestFramework/Microsoft.Rest.ClientRuntime.Azure.TestFramework/TestEnvironment.cs (hunk 2 above)
Comment (shahabhijeet) [inline src/TestFramework/Microsoft.Rest.ClientRuntime.Azure.TestFramework/TestEnvironment.cs]: @markcowl Test has already been added, see comment below.

File: src/TestFramework/Microsoft.Rest.ClientRuntime.Azure.TestFramework/TestEndpoints.cs (hunk 1 above)
Comment (shahabhijeet) [inline src/TestFramework/Microsoft.Rest.ClientRuntime.Azure.TestFramework/TestEndpoints.cs]: Done

**Review by markcowl — APPROVED** [review]

**Pull request merged**